src/nodes.coffee

```diff
@@ -3367,6 +3398,35 @@ exports.Op = class Op extends Base
         answer = [].concat lhs, @makeCode(" #{@operator} "), rhs
         if o.level <= LEVEL_OP then answer else @wrapInParentheses answer
 
+  astType: ->
+    switch @operator
+      when 'new'           then 'NewExpression'
+      when '||', '&&', '?' then 'LogicalExpression'
```

Comment:
Had realized that binary `?` really behaved more like a `LogicalExpression` (ie `and`/`or`/`&&`/`||`) than a `BinaryExpression` in that it has similar control flow semantics. And then saw that the proposed JS version of this operator (`??`) does in fact generate a `LogicalExpression` eg in Babel